app/src/main/java/org/mozilla/fenix/ext/NavController.kt

```diff
@@ -10,19 +10,38 @@ import androidx.navigation.NavDirections
 import androidx.navigation.NavOptions
 import io.sentry.Sentry
 import org.mozilla.fenix.components.isSentryEnabled
+import org.mozilla.fenix.perf.waitForNavGraphInflation
 
 /**
  * Navigate from the fragment with [id] using the given [directions].
  * If the id doesn't match the current destination, an error is recorded.
  */
 fun NavController.nav(@IdRes id: Int?, directions: NavDirections, navOptions: NavOptions? = null) {
+
+    waitForNavGraphInflation(this)
+
     if (id == null || this.currentDestination?.id == id) {
         this.navigate(directions, navOptions)
     } else {
         recordIdException(this.currentDestination?.id, id)
     }
 }
 
+fun NavController.loadNavGraphBeforeNavigate(resId: Int) {
```

Comment:
nit: this is a good descriptive name but when naming something that we're overloading, I prefer to postfix the name because I think it's more likely someone will type in the old method name, notice the new one in auto-complete and use it instead. Here are some (maybe) bad naming ideas:
- `navigateFenix`
- `navigateBlockingForAsyncNavGraph`
- `navigateWithNavGraph`

This doesn't need to change but I think it'd be more intuitive.